crates/snapbox/src/assert.rs

```diff
@@ -483,7 +484,20 @@ impl Assert {
     ///
     /// The default is to auto-detect
     pub fn binary(mut self, yes: bool) -> Self {
-        self.binary = Some(yes);
+        self.data_format = if yes {
+            Some(DataFormat::Binary)
+        } else {
+            Some(DataFormat::Text)
+        };
+        self
+    }
+
+    pub fn data_format(&self) -> Option<&DataFormat> {
+        self.data_format.as_ref()
+    }
+
+    pub fn set_data_format(mut self, data_format: DataFormat) -> Self {
+        self.data_format = Some(data_format);
         self
     }
```

Comment:
We shouldn't be exposing this yet, we've not fully worked out how it works within this API